auslib/blobs/base.py

```diff
@@ -55,6 +55,41 @@ def createBlob(data):
     return blob_map[schema_version](**data)
 
 
+def merge_dicts(ancestor, left, right):
+    result = {}
+    for key in ancestor.keys() + left.keys() + right.keys():
+        key_types = set([type(ancestor.get(key)), type(left.get(key)), type(right.get(key))])
+        key_types.discard(type(None))
+        if len(key_types) > 1 and not key_types.issubset([str, unicode]):
+            raise ValueError("Cannot merge blobs: type mismatch for '{}'".format(key.encode('ascii', 'replace')))
+
+        if isinstance(ancestor.get(key), dict) or isinstance(left.get(key), dict) or isinstance(right.get(key), dict):
+            result[key] = merge_dicts(ancestor.get(key, {}), left.get(key, {}), right.get(key, {}))
+        elif isinstance(ancestor.get(key), list) or isinstance(left.get(key), list) or isinstance(right.get(key), list):
+            result[key] = ancestor.get(key, [])[:]
+            for side in (left, right):
+                for i in side.get(key, []):
+                    if i not in result[key] or not isinstance(i, type(result[key][result[key].index(i)])):
+                        result[key].append(i)
+        else:
+            if key in ancestor:
+                if key in left and key in right and ancestor[key] != left[key] and ancestor[key] != right[key]:
+                    raise ValueError("Cannot merge blobs: left and right are both changing '{}'".format(key.encode('ascii', 'replace')))
+                if key in left and ancestor[key] != left.get(key):
+                    result[key] = left[key]
+                else:
```

Comment:
I think you need an `if key in right` here too.

Reply:
I think this works either way (line 76 checks that left or right is changing the key, and line 78 checks if it's left, so the only option left is right) -- but maybe explicit is better here. I think we'd need if/elif/else to do this, otherwise result[key] might end up empty.